MapsetVerifier.Checks/Taiko/Compose/UnrankableFinisherCheck.cs

```diff
@@ -0,0 +1,195 @@
+using MapsetVerifier.Parser.Objects;
+using MapsetVerifier.Parser.Objects.HitObjects;
+using MapsetVerifier.Parser.Objects.HitObjects.Taiko;
+using MapsetVerifier.Parser.Objects.TimingLines;
+using MapsetVerifier.Parser.Statics;
+
+using static MapsetVerifier.Checks.Utils.TaikoUtils;
+
+namespace MapsetVerifier.Checks.Taiko.Compose
+{
+    [Check]
+    public class UnrankableFinisherCheck : BeatmapCheck
+    {
+        private const string Warning = nameof(Issue.Level.Warning);
+        private const string Problem = nameof(Issue.Level.Problem);
+        
+        private readonly Beatmap.Difficulty[] difficulties =
+        [
+            Beatmap.Difficulty.Easy,
+            Beatmap.Difficulty.Normal,
+            Beatmap.Difficulty.Hard,
+            Beatmap.Difficulty.Insane,
+            Beatmap.Difficulty.Expert,
+            Beatmap.Difficulty.Ultra
+        ];
+
+        public override CheckMetadata GetMetadata() =>
+            new BeatmapCheckMetadata()
+            {
+                Author = "Hivie, Nostril",
+                Category = "Compose",
+                Message = "Unrankable finishers",
+                Difficulties = difficulties,
+                Modes = [Beatmap.Mode.Taiko],
+                Documentation = new Dictionary<string, string>()
+                {
+                    {
+                        "Purpose",
+                        @"
+                    Ensuring that finishers abide by each difficulty's Ranking Criteria."
+                    },
+                    {
+                        "Reasoning",
+                        @"
+                    Improper finisher usage can lead to significant gameplay issues."
+                    }
+                }
+            };
+
+        public override Dictionary<string, IssueTemplate> GetTemplates() =>
+            new()
+            {
+                {
+                    Warning,
+                    new IssueTemplate(
+                        Issue.Level.Warning,
+                        "{0} Abnormal finisher, ensure this makes sense",
+                        "timestamp - "
+                    ).WithCause("Finisher is potentially violating the Ranking Criteria")
+                },
+                {
+                    Problem,
+                    new IssueTemplate(
+                        Issue.Level.Problem,
+                        "{0} Unrankable finisher",
+                        "timestamp - "
+                    ).WithCause("Finisher is violating the Ranking Criteria")
+                }
+            };
+
+        public override IEnumerable<Issue> GetIssues(Beatmap beatmap)
+        {
+            var finishers = beatmap.HitObjects
+                .Where(x => x is Circle && x.IsFinisher())
+                .ToList();
+
+            // any finisher pattern spacing equal to or smaller than this gap is a problem
+            var maximalGapBeats = new Dictionary<Beatmap.Difficulty, double>()
```

Comment:
Those maps can be put outside the function scope as they are not BPM-based or anything.
I would move them to the class level instead, so we don't recreate dictionaries for each beatmap when this check runs.

You could even go with constant numbers so we don't have to calculate `1.0/3` multiple times, as that seems a bit unnecessary.